packages/flutter_tools/gradle/src/main/groovy/app_plugin_loader.groovy

```diff
@@ -0,0 +1,38 @@
+import groovy.json.JsonSlurper
+import org.gradle.api.Plugin
+import org.gradle.api.initialization.Settings
+
+apply plugin: FlutterAppPluginLoaderPlugin
+
+class FlutterAppPluginLoaderPlugin implements Plugin<Settings> {
+    @Override
+    void apply(Settings settings) {
+        def flutterProjectRoot = settings.settingsDir.parentFile
+
+        // If this logic is changed, also change the logic in module_plugin_loader.gradle.
+        def pluginsFile = new File(flutterProjectRoot, '.flutter-plugins-dependencies')
+        if (!pluginsFile.exists()) {
+            return
+        }
+
+        def object = new JsonSlurper().parseText(pluginsFile.text)
+        assert object instanceof Map
+        assert object.plugins instanceof Map
+        assert object.plugins.android instanceof List
+        // Includes the Flutter plugins that support the Android platform.
+        object.plugins.android.each { androidPlugin ->
+            assert androidPlugin.name instanceof String
+            assert androidPlugin.path instanceof String
+            // Skip plugins that have no native build (such as a Dart-only implementation
```

Comment:
Great comment.

Reply:
Not mine :) I mostly copy-pasted this haha